pkg/gatherers/workloads/workloads_info.go

```diff
@@ -75,25 +74,56 @@ func gatherWorkloadInfo(
 
 	// load pods in order
 	start := time.Now()
+	limitReached, info, err := workloadInfo(ctx, coreClient, imageCh)
+	if err != nil {
+		return []record.Record{}, []error{err}
+	}
+
+	workloadImageResize(info.PodCount)
+
+	records := []record.Record{
+		{
+			Name: "config/workload_info",
+			Item: record.JSONMarshaller{Object: &info},
+		},
+	}
+
+	close(imageCh)
```

Comment:
I agree, there are a function call before the close statement. It seems good catching expected errors but using defer costs almost nothing and it could safe some future errors.

Reply:
Well, if I try to add `defer` to close the `imageCh` it just doesn't give me the `imageCount` or the `images` at the report.

Reply:
Ohh yes it seems too late, because then there's no time (or a way) to read from the `imagesDoneCh` channel. Perhaps we could close this channel somewhere else.

Reply:
Let me see what I can do.

Reply:
@tremes @ncaak check it now... I am still not happy with it. I tried to defer and create both channels in several places but it was causing panic or it was not generating the correct report data.

Reply:
Why are you not happy with that? I think it's OK

Reply:
I would like to close the channel where it is created, but it doesn't seem possible. I mean, I could be wrong here but, maybe adding some `WaitGroup` would help, but it will require some even bigger refactoring.